src/strife/f_wipe.c

```diff
@@ -75,9 +75,16 @@ wipe_initColorXForm
   int	ticks )
 {
     memcpy(wipe_scr, wipe_scr_start, width*height*sizeof(*wipe_scr));
-    // [crispy] arm fail-safe crossfade counter with
-    // 13 screen transitions, "zero" count won't be used.
-    fade_counter = 14;
+    // [crispy] arm fail-safe crossfade counter with...
+#ifndef CRISPY_TRUECOLOR
+    // 13 screen transitions in paletted render,
+    // to make duration identical to Vanilla Strife.
+    fade_counter = 13;
+#else
+    // 32 screen screen transitions in TrueColor render,
+    // to make blending effect smooth enough.
+    fade_counter = 32;
```

Comment:
I'm not sure why we need factor 3 more steps in the truecolor case. Please remember that the wipe functions are called every frame, but should only advance every tic. So, in the end, both palette and truecolor transitions should last equally long.

Reply:
You absolutely right. Wiping is advanced every tic, not frame, in `wipe_doColorXForm`. We need factor `3` because of [this](https://github.com/fabiangreffrath/crispy-doom/pull/1204/files/0445bea5a137919ba667b06f6a365bad34139888#diff-3cd6c535b7c91340f32edd1c82567e5861b26c2a2dc43eb247f50398ac097538R420) correction. If it will left as vanilla `< 3`, then fading steps will be "slower" and fading effect itself will be far less smooth. I believe it is possible to have exact vanilla duration and much lesser code, but then effect itself will be not that epic. Exactly the same to vanilla, except with cleaner colors. Just take a try by yourself, you will see the difference immediately.

What is you recommendation then? Make fading effect identical to vanilla, just with cleaner TrueColor colors, fail-safe counter and fixed memory leak?

Reply:
Understood. I'll do changes for smallest diff and one one logics/approach then.

Reply:
Thank you! ❤️

Reply:
This ripped out my heart! 😅 

But pardon me, it was all my fault, I'll explain why. You may remember that when TrueColor support was merged into Strife, I was totally unhappy with the crossfade implementation. Knowing the power of TrueColor, I was absolutely sure it could be better. Then I spent two months experimenting and polishing this effect, and finally, when all the work was done and I saw a properly smooth result, free of magic numbers and tricks, I was amazed. So amazed, that though: _"Finally I'm not ashamed to suggest this approach"._ So amazed, that even forgot that TrueColor support for three non-Doom games wasn't about enhancements, it was about TrueColor only.

For now, it should be done as in vanilla, except with better comments and fixed memory leak. Not sure that the case with custom `XLATAB` will ever happen, but it was a bit personal thing, and the fail-safe counter itself is acts for TrueColor as well.